test/basic/test_lightgbm.jl

```diff
@@ -44,37 +61,49 @@ end
     @testset "find_library works with no system lib" begin
 
         # Arrange
-        settings = setup_env()
-        cp(settings["ref_lib_lightgbm_path"], settings["lib_not_on_sys_fixture_path"]) # fake file copied from lightgbm
+        settings, isborked = setup_env()
+        if !isborked
+            cp(settings["linkable_path"], settings["lib_not_on_sys_fixture_path"], force=true, follow_symlinks=true)
```

Comment:
This and its counterpart in the test below seems to be the offending lines for the macOS CI failures:
```
 IOError: open("/usr/lib/libSystem.B.dylib", 0, 0): no such file or directory (ENOENT)
```

Looking at the Apple developer forums, they mention [`dlopen` code potentially failing if file existence is checked](https://developer.apple.com/forums/thread/655588).